### DISCUSSION TITLE
Auto-incremental integer field

### DISCUSSION THREAD

**Discussion by BorisKamp** [directus/directus#23566] [Feature Requests]
### Summary

Hey guys! I don't see a auto-incremental integer field to pick in a model, is there any? I see this can be setup for the id field of a model, but not for any other custom fields on the model. I have a id field which is set to a UUID, next to that I have a display_id field which I want to an auto-incremental integer (like the primary field key option).

### Motivation

Because there are use cases where other fields then the primary key fields need this.

### Detailed Design

When adding a new field to the model, be able to select `auto incremental integer` as type when selecting a input field type.

### Requirements List

n.a.

### Drawbacks

n.a.

### Alternatives

n.a.

### Adoption Strategy

n.a.

**Comment by TBG-FR** [comment]
Did you find something to do it ? I find that abnormal to be forced to manually check/generate/put an id when a field is auto-generated

**Reply by BorisKamp** [reply]
No I found nothing but doing it manually using hooks

**Reply by TBG-FR** [reply]
> No I found nothing but doing it manually using hooks

Oh sorry I've re-read your issue and mine wasn't the same, the "auto-incremental integer (like the primary field key option)" wasn't working when leaving the field empty (at first I got `ID: Value is required`) but then I retried and it works, it takes the next available autoincrement id

If you want to autoincrement another field, I guess you have no other choice than using hooks or editing the DB Schema using a SQL script (something like `AUTO_INCREMENT` for MySQL and  `GENERATED BY DEFAULT AS IDENTITY` for PostgreSQL)
